### PR TITLE
x509-cert: builder: support RandomizedSigner too.

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

File: x509-cert/src/builder.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ pub enum Error {
     /// Public key errors propagated from the [`spki::Error`] type.
     PublicKey(spki::Error),
 
-    /// Signing error propagated for the [`signature::Signer`] type.
+    /// Signing error propagated for the [`signature::Error`] or type.
```
Comment (baloo) [inline x509-cert/src/builder.rs:35]:
```suggestion
    /// Signing error propagated for the [`signature::Error`] type.
```

**Review by baloo — APPROVED** [review]

**Comment by baloo** [comment]
zlint complains about:
```
failed lints: {"e_mp_rsassa-pss_parameters_encoding_in_signature_algorithm_correct": LintStatus { status: Error, details: Some("RSASSA-PSS parameters are not properly encoded. 3 presentations are allowed but got the unsupported 303806092a864886f70d01010a302ba00d300b0609608648016503040201a11a301806092a864886f70d010108300b0609608648016503040201") }}
```

This is in relation to https://www.mozilla.org/en-US/about/governance/policies/security-group/certs/policy/#511-rsa
```
RSASSA-PSS with SHA-256, MGF-1 with SHA-256, and a salt length of 32 bytes.

The encoded AlgorithmIdentifier MUST match the following hex-encoded bytes:

304106092a864886f70d01010a3034a00f300d0609608648016503040201
0500a11c301a06092a864886f70d010108300d0609608648016503040201
0500a203020120

RSASSA-PSS with SHA-384, MGF-1 with SHA-384, and a salt length of 48 bytes.

The encoded AlgorithmIdentifier MUST match the following hex-encoded bytes:

304106092a864886f70d01010a3034a00f300d0609608648016503040202
0500a11c301a06092a864886f70d010108300d0609608648016503040202
0500a203020130

RSASSA-PSS with SHA-512, MGF-1 with SHA-512, and a salt length of 64 bytes.

The encoded AlgorithmIdentifier MUST match the following hex-encoded bytes:

304106092a864886f70d01010a3034a00f300d0609608648016503040203
0500a11c301a06092a864886f70d010108300d0609608648016503040203
0500a203020140
```

**Comment by lumag** [comment]
@baloo yes, this will be fixed once the `rsa` crate is updated. See the changes at  https://github.com/RustCrypto/RSA/pull/299 and https://github.com/RustCrypto/formats/pull/1010

**Comment by baloo** [comment]
ha, oh. got it.

**Comment by lumag** [comment]
@tarcieri @baloo should be fixed now

**Review by baloo — APPROVED** [review]

**Pull request merged**